### DISCUSSION TITLE
Splunk source health check URL.

### DISCUSSION THREAD

**Discussion by hlbraddock** [vectordotdev/vector#12097] [Q&A] [answered]
We're configuring https://vector.dev/docs/reference/configuration/sources/splunk_hec source running in a Docker (timberio/vector) container.
We need to identify a health check URL.
Our infrastructure has a limitation that the Docker container (AWS ECS) running vector.dev can only expose one port.
Does https://vector.dev/docs/reference/configuration/sources/splunk_hec provide a no-auth http GET based health check URL?
Does vector.dev running in the Docker container (timberio/vector) provide a no-auth http GET based health check URL accessible via the same port (as Splunk source)?

**Comment by jszwedko** [comment, marked as answer]
Hi @hlbraddock !

The `splunk_hec` source exposes a health check at `/health` or `/health/1.0`. This endpoint does not require authentication.

**Reply by hlbraddock** [reply]
Yes, @jszwedko you are correct, that URL `/services/collector/health/1.0`  is supported but (unfortunately, based on my testing), it requires authorization :( 

I get a HTTP/1.1 400 Bad Request response **without** `-H "Authorization: Splunk ${SPLUNK_HEC_TOKEN}"`

Are you able to receive 200 response **without** authorization?

Is there something I should be doing differently?

**Reply by jszwedko** [reply]
Ah, yes, you are right, it does require authorization:

https://github.com/vectordotdev/vector/blob/3536fb956d045fd8d240f98fba290b65e0f8604a/src/sources/splunk_hec/mod.rs#L366-L387

This seems to be the required by the Splunk HEC API as documented in https://docs.splunk.com/Documentation/Splunk/8.2.5/RESTREF/RESTinput#services.2Fcollector.2Fhealth since it documents that it returns a 400 for invalid HEC token.

Vector also exposes a health check over its API (`api.enabled = true`) at `/health`, but this would be on a separate port. Is it possible to configure ECS to forward both ports?

**Reply by hlbraddock** [reply]
Yes, you're right. And I've tested that locally in a docker container, where I can support multiple ports per docker container.
And, I believe AWS ECS can support multiple ports per container.
However, unfortunately, our internal infrastructure deployment configuration does not (currently) support deploying containers with multiple ports.

**Reply by hlbraddock** [reply]
Any other suggestions? Would there be a health check URL 'elsewhere' (not necessarily associated with the Splunk source) but with the **same** port. (I'm guessing not).

Would I be able to submit a PR with support for no-auth health check URL **within** Splunk source? Or would that not be allowed due to not being part of the Splunk spec?

**Reply by jszwedko** [reply]
Actually, I ran Splunk locally just now and it seems like it may not actually require an authentication token for the health check. I passed an invalid token and it ignored it. I'll open a pull request to remove this restriction from the `splunk_hec` source.

**Reply by hlbraddock** [reply]
Interesting, as I'm still receiving `HTTP/1.1 400 Bad Request` response when passing invalid auth token.

Thanks for the offer to open a PR!

**Reply by jszwedko** [reply]
Ah, with Splunk itself you mean? Or with Vector? I tried running Splunk 8 locally and it did not seem to do any validation of the HEC token.

**Reply by hlbraddock** [reply]
I'm referring specifically to https://vector.dev/docs/reference/configuration/sources/splunk_hec/

**Reply by jszwedko** [reply]
Gotcha, yeah, I see. I think the `splunk_hec` source should also ignore the token so I'll make that change. Thanks for raising this issue!

**Reply by hlbraddock** [reply]
Thank you!

**Reply by jszwedko** [reply]
https://github.com/vectordotdev/vector/pull/12098

**Reply by hlbraddock** [reply]
I note the [test](https://github.com/vectordotdev/vector/pull/12098/files#diff-6338dcc8d0f2cf66947ca94a6816250b9fb5f94864fcb61fecaaa45180e55cddR1466) still sends a header, even if invalid. 

Is the header itself still required for the health check? 

If so, https://github.com/vectordotdev/vector/pull/12098 may not resolve our issues, as ECS doesn't support sending headers with health check URL's.

**Reply by jszwedko** [reply]
Ah, nah, it doesn't require the header at all. I just added a test for an invalid token, but no header at all would also work. I'll add a separate test for that.

**Reply by hlbraddock** [reply]
[Yahoo!](https://github.com/vectordotdev/vector/pull/12098/files#diff-6338dcc8d0f2cf66947ca94a6816250b9fb5f94864fcb61fecaaa45180e55cddR1474-R1485) Thanks.

What does your deployment model/schedule look like? When would we expect https://github.com/vectordotdev/vector/pull/12098 made available in say https://hub.docker.com/r/timberio/vector?

**Reply by jszwedko** [reply]
It will be in the next nightly release (the images tagged with dates), but also in the next stable release (which is targeted for next week).
